File: Maths/FindOdd.cxx
// Find the Odd Integer
// https://edabit.com/challenge/wgnmQTbfssuhvZHqe

#include <bits/stdc++.h>
using namespace std;

class Solution{
	public:
	static void findOdd(vector<int> arr){
		unordered_map<int,int> hash;
		for(auto &t : arr){
			hash[t]++;
		}
		for(auto &x : hash){
			if(x.second&1)
				cout << x.first << endl;
		}
		return;
	}
};

int main(){
	ios_base::sync_with_stdio(0);
	cout.tie(0);
	vector<int> test = {20,1,-1,2,-2,3,3,5,5,1,2,4,20,4,-1,-2,5};
	Solution::findOdd(test);
	return 0;
}

File: Maths/FindOdd.py
# Find the Odd Integer
# https://edabit.com/challenge/wgnmQTbfssuhvZHqe

from collections import Counter

def findOdd(arr:list):
	h = Counter(arr)
	for (key,val) in h.items():
		if(val%2!=0):
			print(key,end="\n")
	return

if __name__=='__main__':
	test = [20,1,-1,2,-2,3,3,5,5,1,2,4,20,4,-1,-2,5]
	findOdd(test)
